src/main/java/org/testng/internal/Parameters.java

```diff
@@ -735,6 +738,35 @@ public Boolean call() throws Exception {
     }
   }
 
+  /**
+   * Gets an array of parameter values returned by data provider or the ones that
+   * are injected based on parameter type. The method also checks for {@code NoInjection}
+   * annotation
+   *
+   * @param parameterValues parameter values from a data provider
+   * @param method method to be invoked
+   * @param context test context
+   */
+  public static Object[] injectParameters(Object[] parameterValues, Method method, ITestContext context)
+          throws TestNGException {
+    MethodMatcherContext matcherContext = new MethodMatcherContext(method, parameterValues, context, null);
+    final MethodMatcher matcher = new DataProviderMethodMatcher(matcherContext);
+    return matcher.getConformingArguments();
+  }
+
+  public static Object[] getParametersFromIndex(Iterator<Object[]> parametersValues, int index) {
+    while (parametersValues.hasNext()) {
```

Comment:
foreach

Reply:
Same case here as well. I am not able to use a `foreach` on an `Iterator`

Reply:
ok